<launch>
    <!-- Play the bag file with the ROS clock -->
    <node 
        pkg    = "rosbag" 
        type   = "play" 
        name   = "play_bag" 
        args   = "/home/ubuntu/catkin_ws/src/hole_tracker/bags/flight_1.bag --rate 0.5 --clock /home/ubuntu/catkin_ws/src/hole_tracker/bags/flight_1.bag " 
        output = "screen" 
    />

    <!-- Start Tracker Node -->
    <node 
        pkg           = "hole_tracker" 
        type          = "node_tracker.py" 
        name          = "tracker_node" 
        output        = "screen" 
        launch-prefix = "python3" 
    />
    
    <!-- Start Detector Node -->
    <node 
        pkg           = "hole_tracker" 
        type          = "node_detector_yolo.py" 
        name          = "detector_node" 
        output        = "screen" 
        launch-prefix = "python3" 
    />

    <!-- Open RViz with a specific configuration file -->
    <node 
        pkg    = "rviz" 
        type   = "rviz" 
        name   = "rviz" 
        args   = "-d $(find hole_tracker)/rviz/config_1.rviz" 
        output = "screen" 
    />
</launch>



